docs/generating-tests.rst

```diff
@@ -0,0 +1,370 @@
+
+``testeth`` with ``--filltests`` fills every test filler it finds. The command might modify existing test cases. After running ``testeth`` with ``--filltests``\ , try running ``git status`` in the ``tests`` directory. If ``git status`` indicates changes in unexpected files, that is an indication that the behavior of ``Aleth`` changed unexpectedly.
+
+.. note::
+   If ``testeth`` is looking for tests in the ``../../test/jsontests`` directory (falling back to a path relative to the ``Aleth`` build directory if ``ETHEREUM_TEST_PATH`` is not set), you have probably not specified the ``--testpath`` option.
+
+
+Trying the Filled Test
+----------------------
+
+Trying the Filled Test Locally
+++++++++++++++++++++++++++++++
+
+For trying the filled test, in ``aleth/build`` directory, run
+
+.. code:: bash
+
+   ETHEREUM_TEST_PATH="../../tests" test/testeth -t GeneralStateTests/stExample2
+
+
+Trying the Filled Test in Travis CI
++++++++++++++++++++++++++++++++++++
+
+The following instructions are highly specific to the Aleth C++ Ethereum client, which is currently used for test generation. Once a new test generation tool is ready, this process will likely change.
+
+Goal here is the get the ``Aleth`` Travis CI build to run the new tests with ``Aleth`` to check they pass. To do that a PR has to be submitted to Aleth that updates the git submodule for ethereum/tests to point to a branch with the new tests.
+
+Preparations on the ethereum/tests side
+---------------------------------------
+
+For trying the filled test(s) on ``Travis CI`` for ``Aleth``, the new test cases need to exist in a branch in ``ethereum/tests``. For this, ask somebody with a push permission to ``ethereum/tests``.
+
+
+Preparations on the Aleth side
+------------------------------
+
+Enter ``aleth/test/jsontests`` directory, and checkout the new branch in ``ethereum/tests`` as described in the instructions above. Then go back to the main ``Aleth`` directory and perform ``git add test/jsontests`` followed by ``git commit``.
+
+When you file this commit as a pull request to ``Aleth``, Travis CI should try the newly filled tests.
+
+git commit
+----------
+
+After these are successful, the filler file and the filled test should be added to the ``tests`` repository. File these as a pull request.
+
+If changes in the ``Aleth`` code itself were necessary, also file a pull request for these changes.
+
+Advanced: Converting a GeneralStateTest Case into a BlockchainTest Case
+=======================================================================
+
+In the tests repository, each GeneralStateTest is eventually translated into a BlockchainTest.  This can be done by the following sequence of commands.
+
+.. code::
+
+  ETHEREUM_TEST_PATH="../../tests" test/testeth -t GeneralStateTests/stExample2 -- --filltests --fillchain
```

Comment:
I remember reports that relative path does not work like this.

Reply:
Whew, that's a biggie! Have replaced all the inline pre-pended paths commands and added a note on all occurences to have a look at the unified section explaining how to set ``ETHEREUM_TEST_PATH``.